go/mysql/sql_error.go

```diff
@@ -152,10 +152,12 @@ func mapToSQLErrorFromErrorCode(err error, msg string) *SQLError {
 	}
 }
 
-var stateToMysqlCode = map[vterrors.State]struct {
+type mysqlCode struct {
 	num   int
 	state string
-}{
+}
+
+var stateToMysqlCode = map[vterrors.State]mysqlCode{
```

Comment:
This can also be a slice I think since `vterrors.State` is an integer and we start from `0`? Or is there another reason to use a `map` instead?

Reply:
Nothing is preventing us from using a slice here instead. However, I was thinking that it would be easier to maintain than relying on the offset of the map to match the `vterrors.State` value.